cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -87,21 +87,41 @@ def instrument_name(self):
     def instrument_name(self, instrument_name):
         self._instrument_name = instrument_name
 
-    def __init__(self, app, instrument_name=None, par_dic=None, config=None, data_server_call_back=False, verbose=False, get_meta_data=False, download_products=False, resolve_job_url=False):
-        self.logger = logging.getLogger(repr(self))
+    @property
+    def query_progression(self):
+        if not hasattr(self, '_query_progression'):
+            self._query_progression = []
+        
+        return self._query_progression
+
+    def log_query_progression(self, message):        
+        self.query_progression.append(dict(
+                t_s = time_.time(),
+                message = message,                
+            ))
+
+        t0 = self.query_progression[0]['t_s']
+        self.logger.warning("%s %s s", message, self.query_progression[-1]['t_s'] - t0)
```

Comment:
Is this intended just for being used inside the `InstrumentQueryBackend` class?

Reply:
yes, and it is used.